src/payment/strategies/chasepay/chasepay-payment-strategy.ts

```diff
@@ -0,0 +1,209 @@
+            signInButton,
+            container,
+        } = chasepayCheckoutOptions;
+
+        return this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(methodId))
+            .then(state => {
+                this._paymentMethod = state.paymentMethods.getPaymentMethod(methodId);
+
+                if (!this._paymentMethod || !this._paymentMethod.initializationData) {
+                    throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+                }
+
+                const cart = state.cart.getCart();
+                const storeConfig = state.config.getStoreConfig();
+
+                this._wepayRiskClient.initialize();
+
+                if (!cart || !storeConfig || !this._paymentMethod) {
+                    throw new MissingDataError(MissingDataErrorType.MissingCart);
+                }
+
+                return this._chasePayScriptLoader.load(this._paymentMethod.config.testMode)
+                    .then(({ ChasePay }) => {
+                        const { COMPLETE_CHECKOUT, CANCEL_CHECKOUT } = ChasePay.EventType;
+
+                        if (ChasePay.isChasePayUp && document.getElementById(container)) {
+                            ChasePay.insertBrandings({
+                                containers: [container],
+                                color: 'white',
+                            });
+                        }
+
+                        ChasePay.on(CANCEL_CHECKOUT, () => onCancel());
+
+                        ChasePay.on(COMPLETE_CHECKOUT, (payload: ChasePaySuccessPayload) => {
+                            const state = this._store.getState();
+                            const method = state.paymentMethods.getPaymentMethod(methodId);
+                            const requestId = method && method.initializationData && method.initializationData.merchantRequestId;
+
+                            this._setExternalCheckoutData(payload, requestId).then(() => {
+                                this._paymentInstrumentSelected().then(() => onSuccess(payload));
+                            });
+                        });
+
+                        ChasePay.on(ChasePay.EventType.START_CHECKOUT, () => {
+                            this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(methodId))
+                                .then(() => {
+                                    const state = this._store.getState();
+                                    const method = state.paymentMethods.getPaymentMethod(methodId);
+                                    const sessionId = method && method.initializationData && method.initializationData.digitalSessionId;
+
+                                    if (sessionId) {
+                                        ChasePay.startCheckout(sessionId);
+                                    }
+                                });
+                        });
+
+                        if (signInButton) {
```

Comment:
I think what you should do instead in the `initialize` step is to register event callbacks so that the strategy knows what to do when the shopper clicks on the sign in or edit card button. i.e.:

```js
checkoutService.initializePayment({
    methodId: 'chasepay',
    chasepay: {
        editLink: '#foo',
        signInLink: '#bar',
    },
});
```

Reply:
But if we do something like this, all the logic of the `chasepay` lightbox will be on `ng-checkout` which is the whole  purpose of this user story, so continue working on this PR will be unnecessary , don't you think?

Reply:
Why do you think that way? I think it might be better if I discuss with you directly on this matter.